src/test/java/com/puppycrawl/tools/checkstyle/PackageObjectFactoryTest.java

```diff
@@ -19,57 +19,69 @@
 
+import static com.puppycrawl.tools.checkstyle.PackageObjectFactory.BASE_PACKAGE;
+import static com.puppycrawl.tools.checkstyle.PackageObjectFactory.CHECK_SUFFIX;
+import static com.puppycrawl.tools.checkstyle.PackageObjectFactory.NULL_LOADER_MESSAGE;
+import static com.puppycrawl.tools.checkstyle.PackageObjectFactory.NULL_PACKAGE_MESSAGE;
+import static com.puppycrawl.tools.checkstyle.PackageObjectFactory.PACKAGE_SEPARATOR;
+import static com.puppycrawl.tools.checkstyle.PackageObjectFactory.STRING_SEPARATOR;
+import static com.puppycrawl.tools.checkstyle.PackageObjectFactory.UNABLE_TO_INSTANTIATE_EXCEPTION_MESSAGE;
+import static org.junit.Assert.assertEquals;
 import static org.junit.Assert.assertFalse;
 import static org.junit.Assert.assertNotNull;
 import static org.junit.Assert.fail;
+import static org.mockito.Mockito.mock;
+import static org.mockito.Mockito.when;
 
 import java.io.File;
+import java.io.IOException;
 import java.lang.reflect.Field;
 import java.lang.reflect.Method;
+import java.net.URLClassLoader;
 import java.util.Collection;
 import java.util.Collections;
 import java.util.HashSet;
 import java.util.List;
 import java.util.Map;
 import java.util.Set;
 
-import org.junit.Assert;
 import org.junit.Test;
 
 import com.puppycrawl.tools.checkstyle.api.AbstractFileSetCheck;
 import com.puppycrawl.tools.checkstyle.api.CheckstyleException;
+import com.puppycrawl.tools.checkstyle.api.LocalizedMessage;
 import com.puppycrawl.tools.checkstyle.checks.naming.ConstantNameCheck;
 import com.puppycrawl.tools.checkstyle.internal.CheckUtil;
 
 /**
  * Enter a description of class PackageObjectFactoryTest.java.
+ *
  * @author Rick Giles
  */
 public class PackageObjectFactoryTest {
 
     private final PackageObjectFactory factory = new PackageObjectFactory(
-        new HashSet<>(), Thread.currentThread().getContextClassLoader());
+            BASE_PACKAGE, Thread.currentThread().getContextClassLoader());
 
     @Test
-    public void testCtorException() {
-        try {
+    public void testCtorNullLoaderException() {
+        assertException(IllegalArgumentException.class, NULL_LOADER_MESSAGE, () -> {
             new PackageObjectFactory(new HashSet<>(), null);
-            fail("Exception is expected");
-        }
-        catch (IllegalArgumentException ex) {
-            Assert.assertEquals("moduleClassLoader must not be null", ex.getMessage());
-        }
+        });
+        assertException(IllegalArgumentException.class, NULL_LOADER_MESSAGE, () -> {
+            new PackageObjectFactory("test", null);
+        });
     }
 
     @Test
-    public void testCtorException2() {
-        try {
-            new PackageObjectFactory("test", null);
-            fail("Exception is expected");
-        }
-        catch (IllegalArgumentException ex) {
-            Assert.assertEquals("moduleClassLoader must not be null", ex.getMessage());
-        }
+    public void testCtorNullPackageException() {
```

Comment:
I feel we are not gaining anything but having harder to read tests by having this utility method and doing weird exception checks.
remove `assertException` and split these into 2 different tests. Hard code the `assertException` into each test, and catch the desired exception.